src/R/Components/Impl/PackageManager/Implementation/ViewModel/RPackageManagerViewModel.cs

```diff
@@ -438,30 +412,37 @@ internal class RPackageManagerViewModel : BindableBase, IRPackageManagerViewMode
         }
 
         public async Task SwitchToInstalledPackagesAsync() {
-            await _coreShell.SwitchToMainThreadAsync();
-            if (_selectedTab == SelectedTab.InstalledPackages) {
-                return;
+            if (await SetTabAsync(SelectedTab.InstalledPackages)) {
```

Comment:
Since it is a public method, I'd like to have `await _coreShell.SwitchToMainThreadAsync();` in the beginning of it anyway